pkg/cli/admin/release/extract_tools.go

```diff
@@ -320,20 +328,30 @@ func (o *ExtractOptions) extractCommand(command string) error {
 			missing.Insert(target.Mapping.Image)
 			continue
 		}
-		klog.V(2).Infof("Will extract %s from %s", target.Mapping.From, spec)
-		ref, err := imagereference.Parse(spec)
+		// Always try the registry/repo passed from the user first.  If image not found,
+		// proceed with the release info from the image spec
+		desired, err := imagereference.Parse(o.From)
 		if err != nil {
 			return err
 		}
-		target.Mapping.Image = spec
-		target.Mapping.ImageRef = imagesource.TypedImageReference{Ref: ref, Type: imagesource.DestinationRegistry}
-		if target.AsArchive {
-			willArchive = true
-			target.Mapping.Name = fmt.Sprintf(target.ArchiveFormat, releaseName)
-			target.Mapping.To = filepath.Join(dir, target.Mapping.Name)
-		} else {
-			target.Mapping.To = filepath.Join(dir, target.Command)
-			target.Mapping.Name = fmt.Sprintf("%s-%s", target.OS, target.Command)
+		digest, err := imagereference.GetSpecDigest(spec)
+		if err != nil {
+			return err
+		}
+		specFirst := fmt.Sprintf("%s@%s", desired.AsRepository(), digest)
```

Comment:
This is wrong and might cause problems in the future, I'd prefer to use `ref.Exact()` instead or any other for printing.

Reply:
refExact is something like this: `localhost:5000/ocp/release:4.5.0-0.nightly-2020-04-18-093630` but we need just the `localhost:5000/ocp/release`  and for the tag, we need the sha of 
`localhost:5000/ocp/release:4.5.0-0.nightly-2020-04-18-093630-installer` (or whatever tool you're extracting)
that sha is always the same as it's reference image, in this case, 
`localhost:5000/ocp/release@sha256:22204fa241eaad3f04d88bfa79629d...` that is the same for the reference image (that a disconnected user will not have access to):
`quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:22204fa241eaad3f04d8...`

Reply:
resolving this, as i've moved all changes to pkg/cli/admin/release/info